crates/metastore/src/database.rs

```diff
@@ -1033,6 +1033,39 @@ impl State {
                 // Update the new storage size
                 self.deployment.storage_size = update_deployment_storage.new_storage_size;
             }
+            Mutation::CreateFunction(f) => {
+                let schema_id = self.get_schema_id(DEFAULT_SCHEMA)?;
+                let oid = self
+                    .schema_objects
+                    .get(&schema_id)
+                    .and_then(|objs| objs.functions.get(&f.name))
+                    .copied()
+                    .unwrap_or_else(|| self.next_oid());
+
+                let ent = FunctionEntry {
+                    meta: EntryMeta {
+                        entry_type: EntryType::Function,
+                        id: oid,
+                        parent: schema_id,
+                        name: f.name.clone(),
+                        builtin: false,
+                        external: true,
+                        is_temp: false,
+                    },
+                    func_type: f.function_type,
+                    signature: Some(f.signature),
+                    user_defined: true,
+                };
+                if let Some(ent) = self.entries.get(&oid)? {
+                    return Err(MetastoreError::DuplicateName(ent.get_meta().name.clone()));
+                }
```

Comment:
(minir) - would it be possible/preferable to move this logic into lines 1038-1043?

It reads to me that we try to get an oid for the function name and otherwise just set the next oid, but maybe I am misunderstanding from reading the code, and this line is checking against the entries map, so maybe it's not the same thing?